input/tcp.go

```diff
@@ -85,21 +87,29 @@ func (s *TCP) Run(inch chan<- *baker.Data) error {
 	}
 	defer l.Close()
 
+	wg := sync.WaitGroup{}
 	for atomic.LoadInt64(&s.stop) == 0 {
 		l.SetDeadline(time.Now().Add(1 * time.Second))
 		conn, err := l.AcceptTCP()
 		if err != nil {
-			if opErr, ok := err.(*net.OpError); ok && opErr.Timeout() {
+			if errors.Is(err, os.ErrDeadlineExceeded) {
 				continue
 			}
 			ctxLog.WithFields(log.Fields{"error": err}).Error("Error while accepting")
 		}
 
 		ctxLog.WithFields(log.Fields{"addr": conn.RemoteAddr()}).Info("Connected")
+
 		wg.Add(1)
 		go func(conn *net.TCPConn) {
-			defer wg.Done()
-			s.handleStream(conn)
+			defer func() {
+				conn.Close()
+				wg.Done()
+			}()
+
+			if err := s.handleStream(conn); err != nil {
+				ctxLog.WithError(err).WithFields(log.Fields{"error": err}).Error("Error when handling stream")
```

Comment:
lowercase?
```suggestion
				ctxLog.WithError(err).WithFields(log.Fields{"error": err}).Error("error when handling stream")
```

Reply:
what must remain lowercases are error strings, since they can  be concatenated / wrapped within others. However here it's a log string, log strings should start with caps